platforms/software/testing-base/src/main/java/org/gradle/api/tasks/testing/TestEventReporterFactory.java

```diff
@@ -48,5 +50,5 @@ GroupTestEventReporter createTestEventReporter(
         String rootName,
         Directory binaryResultsDirectory,
         Directory htmlReportDirectory
-    );
+    ) throws IOException;
```

Comment:
I don't think we should make these exceptions public. The same thing as close(). What would we expect users to do about it?

I think we can just rethrow these as UncheckedIOExceptions.

Reply:
I suppose there's not much for them to do, most likely they'll just rethrow.